.circleci/continue_config.yml

```diff
@@ -96,469 +236,570 @@ jobs:
-            JAVA17_HOME='/usr/lib/jvm/java-17-openjdk-amd64'
-
-            echo 'export JAVA17_HOME=/usr/lib/jvm/java-17-openjdk-amd64' >> "$BASH_ENV"
-            echo 'export SPARK_VERSION_VAR='${SPARK} >> "$BASH_ENV"
-            echo 'export JAVA_BIN='$([ "$JAVA_VERSION_NUMBER" = "17" ] && echo "$JAVA17_HOME/bin/java" || echo "$JAVA8_HOME/jre/bin/java") >> "$BASH_ENV"
-            echo 'export JAVAC_BIN='$([ "$JAVA_VERSION_NUMBER" = "17" ] && echo "$JAVA17_HOME/bin/javac" || echo "$JAVA8_HOME/bin/javac") >> "$BASH_ENV"
       - restore_cache:
           keys:
-            - v1-integration-spark-{{ .Branch }}-{{ checksum "/tmp/checksum.txt" }}
-            - v1-integration-spark-{{ .Branch }}
-      - attach_workspace:
-          at: ~/.m2/repository/io/openlineage/
-      - run: |
-          echo "Setting default Java to ${JAVA_BIN}"
-          sudo update-alternatives --set java ${JAVA_BIN}
-          sudo update-alternatives --set javac ${JAVAC_BIN}
-      - run: ./gradlew --no-daemon --console=plain --stacktrace build -Pspark.version=${SPARK_VERSION_VAR} -Pscala.binary.version=2.13 -Pjava.compile.home=${JAVA17_HOME}
-      - run:
-          when: on_fail
-          command: cat build/test-results/test/TEST-*.xml
-      - run: ./gradlew --no-daemon --console=plain jacocoTestReport -Pscala.binary.version=2.13 -Pjava.compile.home=${JAVA17_HOME}
-      - run: ./gradlew --no-daemon --console=plain --info check -x test -Pspark.version=${SPARK_VERSION_VAR} -Pscala.binary.version=2.13 -Pjava.compile.home=${JAVA17_HOME}
-      - run: ./gradlew --console=plain javadoc -Pscala.binary.version=2.13 -Pjava.compile.home=${JAVA17_HOME}
+            - v1-client-java-{{ .Branch }}-{{ checksum "/tmp/checksum.txt" }}
+            - v1-client-java-{{ .Branch }}
+      - run: ./gradlew --no-daemon --console=plain --stacktrace build
+      - run: ./gradlew --no-daemon --console=plain jacocoTestReport
+      - run: ./gradlew --no-daemon --console=plain --info check
+      - run: bash <(curl -s https://codecov.io/bash)
+      - run: ./gradlew --console=plain javadoc
+      - run: ./gradlew --console=plain publishToMavenLocal --info
+      - run: ls -halt ~/.m2/repository/io/openlineage/openlineage-java
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - .m2/repository/io/openlineage/openlineage-java
+            - openlineage/client/java/build-cache
+            - openlineage/client/java/build
       - store_test_results:
-          path: build/test-results/test
+          path: client/java/build/test-results/test
       - store_artifacts:
           path: build/reports/tests/test
           destination: test-report
       - store_artifacts:
           path: build/reports/tests/pmd
           destination: pmd-report
       - save_cache:
-          key: v1-integration-spark-{{ .Branch }}-{{ checksum "/tmp/checksum.txt" }}
+          key: v1-client-java-{{ .Branch }}-{{ checksum "/tmp/checksum.txt" }}
```

Comment:
That's actually a problem that a whole `.gradle` is saved per branch checksum and stored for 14 days. 
Worth having a followup issue at least.